### PR TITLE
Add backup common functions and types

### PR DESCRIPTION
**Pull request by derekbit**
The common functions and types were originally in longhorn-engine.

https://github.com/longhorn/longhorn/issues/6514

### CODE REVIEW

**Review by ChanYiLin — APPROVED** [review]
LGTM

File: list.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ type VolumeInfo struct {
 
 	BackingImageName     string
 	BackingImageChecksum string
-	StorageClassname     string
```
Comment (shuo-wu) [inline list.go:29]: Will this affect the marshaled json output and cause a compatibility issue?

Comment (derekbit) [reply list.go:29]: Oh... you're right.

Comment (derekbit) [reply list.go]: Updated. Thank you.

**Review by shuo-wu — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@derekbit see if the dependant component repos need to do backport too or not. This change should be used by the master, 1.5 and 1.4 of them.

**Pull request merged**